### ISSUE TITLE
Disable "Run Visual Studio formatting" only for a certain file type

### ISSUE THREAD

**Issue by monoblaine** [codecadwallader/codemaid#234]
When I save a file, I'd like to have CodeMaid do all the cleanup actions with an exception: *If the file is a JavaScript file, do **not** format the file*. I just want to disable the formatting action, but others should remain (It should still add end of file trailing new line, for instance) Is this possible?

**Comment by codecadwallader** [comment]
No, we don't currently have that level of configuration available.  Is there a specific aspect of format document you are trying to avoid or is it a blanket request?

**Comment by monoblaine** [comment]
Visual Studio removes the space between function name and the parentheses:

```js
function hede () {}
```

becomes

```js
function hede() {}
```

And there's no option to disable that.

**Comment by codecadwallader** [comment]
Understood.  It's not complex code to write to add this specific option for your use case, but the difficulty we run into is the complexity of visualizing all of the specific options (and combinations of them) within the Options dialog.  We currently have ~140 options and I'm not sure how to add this one cleanly without considering a bit of redesign.  I'm open to suggestions.

**Comment by monoblaine** [comment]
As far as I understand, options that concern files beyond .NET-related ones are:

* Run format document
* Insert end of file trailing new line
* Remove end of file trailing new line
* Remove end of line whitespace
* Remove multiple consecutive blank lines
* The "File Header" section under `Cleaning > Update`

How about showing a button with a label "Applied To" near all these *universal* (not .NET specific) options? When clicked, display a form with two radio buttons:

* [ ] All the file types selected in `Cleaning > File Types`
* [x] Let me decide (regex)
    `\.(htm|html|cs|less|js|config|asax|ashx|aspx|resx|css|scss|sql|json|txt|md|cshtml)$`
